src/main/java/core/basesyntax/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    private final Storage storage;
+
+    public PurchaseOperation(Storage storage) {

```

Comment:
The constructor of `PurchaseOperation` should not take `Storage` as a parameter. This creates a tight coupling with the `Storage` class. Instead, consider passing the necessary data in the `apply` method or using a service to interact with the storage.